python/cudf/cudf/core/series.py

```diff
@@ -783,39 +785,25 @@ def __len__(self):
         return len(self._column)
 
     def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
-        if method == "__call__" and hasattr(cudf, ufunc.__name__):
-            func = getattr(cudf, ufunc.__name__)
-            return func(*inputs)
+        if method == "__call__":
+            return get_appropriate_dispatched_func(
+                cudf, cupy, ufunc, inputs, kwargs
```

Comment:
FYI @galipremsagar I had to add both `cuDF`, `cuDF.series` as fallback options. 

I had earlier incorrectly switched this to`cudf.Series` from `cudf` and that led to some failures with tests like subtraction b/w `numpy` DateTime scalar and `cudf.Series` .   

Now we will first search `cudf` function space followed by `cudf.Series` function space, finally falling back to `cupy`.  Please let me know what you think. 

Previously we only looked at `cudf` name space for `ufunc` and `cudf.Series` for `array function`. IMO this should be a better solution.